### ISSUE TITLE
Passing the wrong labels for the test set to evaluator

### ISSUE THREAD

**Issue by ArlindKadra** [automl/Auto-PyTorch#279]
https://github.com/automl/Auto-PyTorch/blob/3413bc32758b4be47b4511a09b64aff4f3a8e005/autoPyTorch/evaluation/train_evaluator.py#L305-L319

Shouldn't line 319 be `self.y_test` ?

**Comment by ArlindKadra** [comment]
Additionally, should `opt_pred` not be passed `test_indices `

**Comment by ravinkohli** [comment]
no actually this is ony passed to ensure that the output is in the correct shape.

**Comment by nabenabe0928** [comment]
I checked this issue now, and I got what Ravin says.
But it is confusing, so we should fix the keyword for this argument.

**Comment by ArlindKadra** [comment]
Yeah, also what Ravin is saying is not that correct actually. It does not check if the output is in a correct shape, if so, just passing `X_test` and `y_train[train_indices]` should fail, since the test set and train set are not equal in size.

**Comment by ravinkohli** [comment]
I think I should have been more specific. For multi-class classification problems, if our predictions are not in the form of (n_samples, n_classes) it will make them that way. It is using y_train to get the classes and to map our predictions. But actually I think it is never used and we may not need it also, so it can be removed, but I'll try this locally and let you know in this issue.
